code/utils_processing.py

```diff
@@ -1,210 +1,160 @@
-# DeepSlide
-# Jason Wei, Behnaz Abdollahi, Saeed Hassanpour
+"""
+DeepSlide
+Contains all functions for processing.
 
-# This file contains all the methods for processing
+Authors: Jason Wei, Behnaz Abdollahi, Saeed Hassanpour
 
-import config
-from utils import *
+Last Modified: November 30, 2019 (Joseph DiPalma)
+"""
+
+import functools
+import itertools
+import math
+import time
+from multiprocessing import (Process, Queue, RawArray)
+from pathlib import Path
+from shutil import copyfile
+from typing import (Callable, Dict, List, Tuple)
 
+import numpy as np
 from PIL import Image
-import cv2
+from imageio import (imsave, imread)
+from skimage.measure import block_reduce
 
-import skimage.measure
-from skimage.transform import rescale, rotate
-from scipy.stats import mode
-from scipy.misc import imsave
-Image.MAX_IMAGE_PIXELS = 1e10
+import config
+from utils import (get_all_image_paths, get_image_names, get_image_paths,
+                   get_subfolder_paths)
 
-###########################################
-#           FILTERING WHITESPACE          #
-###########################################
+Image.MAX_IMAGE_PIXELS = None
 
 
-def is_purple_dot(r, g, b):
-    rb_avg = (r + b) / 2
-    if r > g - 10 and b > g - 10 and rb_avg > g + 20:
-        return True
-    return False
+def is_purple(crop: np.ndarray) -> bool:
+    """
+    Determines if a given portion of an image is purple.
 
+    Args:
+        crop: Portion of the image to check for being purple.
 
-# this is actually a better method than is whitespace, but only if your images are purple lols
-def is_purple(crop):
-    pooled = skimage.measure.block_reduce(
-        crop, (int(crop.shape[0] / 15), int(crop.shape[1] / 15), 1),
-        np.average)
-    num_purple_squares = 0
-    for x in range(pooled.shape[0]):
-        for y in range(pooled.shape[1]):
-            r = pooled[x, y, 0]
-            g = pooled[x, y, 1]
-            b = pooled[x, y, 2]
-            if is_purple_dot(r, g, b):
-                num_purple_squares += 1
-    if num_purple_squares > 100:
-        return True
-    return False
+    Returns:
+        A boolean representing whether the image is purple or not.
+    """
+    pooled = block_reduce(image=crop,
+                          block_size=(crop.shape[0] // 15, crop.shape[1] // 15,
+                                      1),
+                          func=np.average)
+
+    return pooled[(pooled[..., 0] > pooled[..., 1] - 10)
```

Comment:
is there a more clear way to write this?

Reply:
Fixed to be more readable in newest commit.